### PR TITLE
[Typing][B-35] Add type annotations for `python/paddle/amp/debugging.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
B35

### CODE REVIEW

File: python/paddle/amp/debugging.py [hunk 1]
```diff
@@ -397,12 +429,12 @@ def check_numerics(
...
 
 
-def _get_operator_stats_flag():
+def _get_operator_stats_flag() -> Any:
     flags = paddle.get_flags(["FLAGS_low_precision_op_list"])
     return flags["FLAGS_low_precision_op_list"]
 
 
-def _print_operator_stats(op_count_dict):
+def _print_operator_stats(op_count_dict: dict[str, str | list[int]]) -> str:
```
Comment (megemini) [inline python/paddle/amp/debugging.py]:
```suggestion
def _print_operator_stats(op_count_dict: dict[str, str | list[int]]) -> None:
```

**Review by megemini — COMMENTED** [review]
不清楚为啥 CI 挂了 ... ... 

贴一下错误信息:

``` shell
2024-07-17 17:39:41 >>> 1 sample codes ran timeout or error in env: {'gpu', 'cpu'}
2024-07-17 17:39:41 paddle.amp.debugging.enable_operator_stats_collection:1, running time: 10.000s


2024-07-17 17:39:45 0   paddle::pybind::eager_api_conv2d(_object*, _object*, _object*)
2024-07-17 17:39:45 1   conv2d_ad_func(paddle::Tensor const&, paddle::Tensor const&, std::vector<int, std::allocator<int> >, std::vector<int, std::allocator<int> >, std::string, std::vector<int, std::allocator<int> >, int, std::string)
2024-07-17 17:39:45 2   conv2d_ad_func(paddle::Tensor const&, paddle::Tensor const&, std::vector<int, std::allocator<int> >, std::vector<int, std::allocator<int> >, std::string, std::vector<int, std::allocator<int> >, int, std::string)
2024-07-17 17:39:45 3   paddle::experimental::conv2d(paddle::Tensor const&, paddle::Tensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::string const&, std::vector<int, std::allocator<int> > const&, int, std::string const&)
2024-07-17 17:39:45 4   void phi::ConvCudnnKernel<phi::dtype::float16, phi::GPUContext>(phi::GPUContext const&, phi::DenseTensor const&, phi::DenseTensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::string const&, std::vector<int, std::allocator<int> > const&, int, std::string const&, phi::DenseTensor*)
2024-07-17 17:39:45 5   void phi::ConvCudnnKernelImplV7<phi::dtype::float16, phi::GPUContext>(phi::DenseTensor const*, phi::DenseTensor const*, phi::GPUContext const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, phi::backends::gpu::DataLayout, phi::backends::gpu::DataLayout, bool, bool, int, phi::DenseTensor*)
2024-07-17 17:39:45 6   std::_Function_handler<void (void*), phi::ConvRunner<phi::dtype::float16, (phi::ConvKind)1>::Apply(phi::GPUContext const&, phi::ConvArgsBase<cudnnContext*, cudnnDataType_t> const&, phi::SearchResult<cudnnConvolutionFwdAlgo_t> const&, phi::dtype::float16 const*, phi::dtype::float16 const*, phi::dtype::float16*, int, int, int, int, unsigned long, phi::DnnWorkspaceHandle*, bool)::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&)
2024-07-17 17:39:45 
2024-07-17 17:39:45 ----------------------
2024-07-17 17:39:45 Error Message Summary:
2024-07-17 17:39:45 ----------------------
2024-07-17 17:39:45 FatalError: `Termination signal` is detected by the operating system.
2024-07-17 17:39:45   [TimeInfo: *** Aborted at 1721209181 (unix time) try "date -d @1721209181" if you are using GNU date ***]
2024-07-17 17:39:45   [SignalInfo: *** SIGTERM (@0xa6) received by PID 277 (TID 0x7fa9450ba740) from PID 166 ***]
2024-07-17 17:39:45  
```

merge 一下重新跑跑？

File: python/paddle/amp/debugging.py (hunk 1 above)
Comment (enkilee) [reply python/paddle/amp/debugging.py]: 收到

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**